### DISCUSSION TITLE
Error: Data does not show when pass onto another component

### DISCUSSION THREAD

**Discussion by Jennelle186** [supabase/supabase#18200] [Questions] [answered]
Also in: https://stackoverflow.com/questions/77292111/error-in-passing-another-component-data-does-not-show-when-passed-onto-another

Description: I have an application with multiple components for displaying and editing water types. I'm using Next.js and Supabase. In my `page.tsx`, I display a list of water types and allow users to edit them by clicking on a link. In my `/app/waterTypes/edit/[id]/page.tsx`, I can see the data in the console, but I'm unsure if I still need the WaterTypeEditForm component from `/app/waterType/edit/editForm.tsx`. I'm facing issues with rendering "No water types available" and need guidance on how to structure my code more efficiently. I am using NextJS13 and Supabase for the auth and db. 

I have this `app/waterTypes/page.tsx` where I retrieve and show all of the list of water. I am using the `Link query` to pass it unto the 

    import { createServerComponentClient } from "@supabase/auth-helpers-nextjs";
    import { cookies } from "next/headers";
    import Link from "next/link";
    import { redirect } from "next/navigation";
    
    export const revalidate = 0;
    
    const WaterTypes = async () => {
        const supabase = createServerComponentClient({ cookies });
        const {data: {session }} = await supabase.auth.getSession();
        if(!session){
          redirect('/login')
        }
    
        const {data: water_types} = await supabase.from("water_type").select()
    
        if(!water_types){
          return <p>No Water Types Found</p>
        }
    
    
        return (
          <div>
            <h1>List of Water Types</h1>
            <Link href='/waterTypes/new'>
                + New bike
            </Link>
            {water_types.map((water) => (
              <ul key={water.id}>
                <li>Type of Water: {water.name}</li>
            <li>Price: {water.price}</li>
    
            <Link
              href={{
                pathname: `/waterTypes/edit/${water.id}`,
                query: water // the data
              }}
            >
              Edit  
            </Link>
    
            <button>
              Delete
            </button>
    
              </ul>
            ))}
          </div>
    
        )
    }
    export default WaterTypes;

And I have this on my `/app/waterTypes/edit/[id]/page.tsx`. In this part, I can see the data on the console. 

    import { createServerComponentClient } from "@supabase/auth-helpers-nextjs";
    import { cookies } from "next/headers";
    import WaterTypeEditForm from "../editForm";
    
    const editWaterTypes = async ({searchParams: water_types} : {searchParams: {
        name: string,
        price: number,
        id: number
    }}) => {
    
        const supabase = createServerComponentClient({ cookies })
    
      const {
        data: { session },
      } = await supabase.auth.getSession()
    
      console.log(water_types,"water types in the page.tsx")
    
        return <WaterTypeEditForm session={session} water_types={water_types}/>
    }
     
    export default editWaterTypes;

And then I have another **WaterTypeEditForm** from `/app/waterType/edit/editForm.tsx`. On this part, I am unsure whether I still need this or if I can just perform the edit function on the `[id]/page.tsx`. In here, it just renders `No water types available.`

    'use client'
    
    import { createClientComponentClient, Session } from "@supabase/auth-helpers-nextjs";
    import { useCallback, useEffect, useState } from "react";
    
    type WaterType = {
        name: string;
        price: number;
        id: number;
    };
    
    export default function WaterTypeEditForm({session, water_types} : { session: Session | null; water_types: WaterType | WaterType[]; } ){
    
        console.log(water_types,"water types in the edit Form")
    
        return(
            <div>Sample
                {Array.isArray(water_types) ? (
                    water_types.map((water) => (
                    <ul key={water.id}>
                        <li>Type of Water: {water.name}</li>
                        <li>Price: {water.price}</li>
                        <button>Delete</button>
                    </ul>
                    ))
                ) : (
                    <p>No water types available.</p>
                )}
            </div>
        )
    }

This is the page structure:

[![enter image description here][1]][1]

  [1]: https://i.stack.imgur.com/xhP1V.png

**Comment by s0h311** [comment, marked as answer]
In your `WaterTypeEditForm` you are accepting `water_types: WaterType | WaterType[]` as prop. Maybe you just have one `WaterType`. And not an Array of it. In this case `Array.isArray(water_types)` would result false.

**Reply by Jennelle186** [reply]
Thank you. I removed the `WaterType[]` and it works now
